src/Zip.Shared/Shared.cs

```diff
@@ -587,38 +587,8 @@ internal static int ReadWithRetry(System.IO.Stream s, byte[] buffer, int offset,
 
             do
             {
-                try
-                {
-                    n = s.Read(buffer, offset, count);
-                    done = true;
-                }
-                catch (System.IO.IOException ioexc1)
-                {
-                    // Check if we can call GetHRForException,
-                    // which makes unmanaged code calls.
-                    var p = new System.Security.Permissions.SecurityPermission(
-                        System.Security.Permissions.SecurityPermissionFlag.UnmanagedCode);
```

Comment:
@mirasrael this class doesn't work but actually I havent tried looking into nuget I will check if I can revert that change but when I looked up is that it isn't recommended to use the SecurityPermission class